model/version_db.go

```diff
@@ -64,27 +64,32 @@ func VersionByIds(ids []string) db.Q {
 // All is a query for all versions.
 var VersionAll = db.Query(bson.D{})
 
-// FindVersionByLastKnownGoodConfig filters on versions with valid (i.e., have no errors) config for the given
-// project. Does not apply a limit, so should generally be used with a findOneRepoRefQ.
+// FindVersionByLastKnownGoodConfig filters on versions with valid (i.e., have no errors) config for the given project.
 func FindVersionByLastKnownGoodConfig(projectId string, revisionOrderNumber int) (*Version, error) {
-	q := bson.M{
-		VersionIdentifierKey: projectId,
-		VersionRequesterKey: bson.M{
-			"$in": evergreen.SystemVersionRequesterTypes,
-		},
-		VersionErrorsKey: bson.M{
-			"$exists": false,
-		},
-	}
+	retryLimit := 50
+	for i := 0; i < retryLimit; i++ {
+		q := bson.M{
+			VersionIdentifierKey: projectId,
+			VersionRequesterKey:  evergreen.RepotrackerVersionRequester,
```

Comment:
I also realized it was maybe a bug to use any other requester here, since git tag versions can technically have any config, not the one the project typically uses. I audited some places we use this and I feel like this change is very safe, but if we're nervous about it I can create a different ticket.